extensions/wikia/CuratedContent/CuratedContentController.class.php

```diff
@@ -354,18 +366,32 @@ public function getData() {
 					$featured[ 'featured' ] = 'true';
 					$curated[] = $featured;
 				}
+
 				$optional = $this->communityDataService->getOptional();
 				if ( !empty( $optional ) ) {
 					$curated[] = $optional;
 				}
 
-				$data = array_map( function ( $section ) {
-					$section[ 'node_type' ] = 'section';
-					$section[ 'items' ] = $this->extendItemsWithImages( $section[ 'items' ] );
-					$section[ 'items' ] = $this->extendItemsWithType( $section[ 'items' ] );
+				$data = array_map( function ( $section ) use ( $validator ) {
```

Comment:
what do you think about extracting these two loops to separate method for clarity ?